### PR TITLE
Relay: sync 'useMutation' with Pipedrive implementation

### PR DESCRIPTION
**Pull request by mrtnzlml**
We not internally need to work with disposables, optimistic responses and we also synced the public interface with Relay Experimental (first the executor, second the pending status).

### CODE REVIEW

File: src/relay/src/hooks/__tests__/useMutation.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+// @flow
+
+/* global document */
+
+import React, { useEffect } from 'react';
```
Comment (jaroslav-kubicek) [inline src/relay/src/hooks/__tests__/useMutation.test.js]: AFAIK we always imported everything `import * as React from 'react';`

Comment (tbergq) [reply src/relay/src/hooks/__tests__/useMutation.test.js]: There is nothing wrong with this import

Comment (jaroslav-kubicek) [reply src/relay/src/hooks/__tests__/useMutation.test.js]: That’s true, I’m trying to point out consistency

Comment (mrtnzlml) [reply src/relay/src/hooks/__tests__/useMutation.test.js]: Feel free to create an Eslint rule if you think it's essential. I think it's better to focus on other things than consistency, which in this case, doesn't bring anything.

**Pull request merged**